assets/js/googlesitekit/widgets/datastore/areas.js

```diff
@@ -160,6 +164,47 @@ export const reducer = ( state, { type, payload } ) => {
 };
 
 export const resolvers = {
+	/**
+	 * Defines default widget areas for a given context.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} contextSlug Widget context to get areas for.
+	 */
+	*getWidgetAreas( contextSlug ) {
```

Comment:
This wasn't really defined in the ACs or IB, but the default widget areas (and later the default widgets) should be defined outside of the datastore. Let's implement this almost as if we were 3P ourselves, using our own API.

I suggest we create an `assets/js/googlesitekit/widgets/register-defaults.js` file with a `registerDefaults( registry )` function that dispatches `registerWidgetArea` and `assignWidgetArea` calls for both `CONTEXT_DASHBOARD` and `CONTEXT_PAGE_DASHBOARD`. This function could then be called from `assets/js/googlesitekit/widgets/index.js` with the `Data` registry.

Reply:
Okay, updated it. Is it better now? I used Widgets API itself instead of using Data registery directly. Hope it's okay.